docs: convert the table from html to md format (#21774)
mdn/content: files/en-us/web/javascript/reference/deprecated_and_obsolete_features/index.md

The following properties are deprecated. This does not affect their use in [replacement strings](/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace):

| Property                                          | Description                                                                                                                                                     |
| ------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | {{jsxref("RegExp/n", "$1-$9")}}                   | Parenthesized substring matches, if any.<br />**Warning:** Using these properties can result in problems, since browser extensions can modify them. Avoid them! |

Reviewer: This is better than the table it replaces, but I'd still be considering a definition list or similar.

Note that the warning format will not work in this context - this just renders as normal markdown without a box. If you keep the table you might do it like this:

> Parenthesized substring matches, if any.<br />_Avoid using as browser extensions can modify them!_

Author: Oh, I just found that this is a warning box mark, but, before the modification of the document, does not show the effect.

> [url](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features#regexp)

Author: The solution:

```suggestion
| {{jsxref("RegExp/n", "$1-$9")}}                   | Parenthesized substring matches, if any.<br />Avoid using these if possible, because browser extensions can modify them!                                        |
```

Reviewer 2: This is not a callout box AFAICT, but yes, I'd still prefer a dl here.

Reviewer 3: > Avoid using these if possible, because browser extensions can modify them

What does that even mean? It sounds extremely improbable that a modern browser extension could modify a native API. Unless we can justify this statement we should remove it.

Reviewer: @YangFong Can you try turning this into a definition list, removing the "because browser extensions can modify them" and adding the comment here: https://github.com/mdn/content/pull/21774#discussion_r1003739173 ?

Sorry to make this bigger than it started, but we might as well do this "right".

Author: How about this:

```suggestion
| {{jsxref("RegExp/n", "$1-$9")}}                   | Parenthesized substring matches, if any.<br />Avoid using these if possible (See also [#21774(comment)](https://github.com/mdn/content/pull/21774#discussion_r1003739173)).|
```

Reviewer 2: Please don't say `See #21774(comment)` in actual MDN prose😅 Especially when the link itself is a message containing a link to elsewhere. You can try to paraphrase the source I've given. If you feel necessary, you can add this as another paragraph below the table, since it applies to other static properties as well.

Author: Sorry, I don't have any good ideas about this change.

```suggestion
| {{jsxref("RegExp/n", "$1-$9")}}                   | Parenthesized substring matches, if any.<br />Avoid using these if possible, since it applies to other static properties as well. |
```

| {{jsxref("RegExp.input", "$_")}}                  | See `input`.                                              |
| {{jsxref("RegExp.lastMatch", "$&amp;")}}          | See `lastMatch`.                                          |
| {{jsxref("RegExp.lastParen", "$+")}}              | See `lastParen`.                                          |
| {{jsxref("RegExp.leftContext", "$`")}}            | See `leftContext`.                                        |
| {{jsxref("RegExp.rightContext", "$'")}}           | See `rightContext`.                                       |
| {{jsxref("RegExp.input", "input")}}               | The string against which a regular expression is matched. |
| {{jsxref("RegExp.lastMatch", "lastMatch")}}       | The last matched characters.                              |
| {{jsxref("RegExp.lastParen", "lastParen")}}       | The last parenthesized substring match, if any.           |
| {{jsxref("RegExp.leftContext", "leftContext")}}   | The substring preceding the most recent match.            |
| {{jsxref("RegExp.rightContext", "rightContext")}} | The substring following the most recent match.            |

> > **Warning:** Avoid using these static properties, as they can cause [a lot of issues when interacting with external code](https://github.com/tc39/proposal-regexp-legacy-features/blob/master/subclass-restriction-motivation.md#legacy-static-properties-regexp1-etc)!

Reviewer: Minor tweak
```suggestion
> **Warning:** Avoid using these static properties, as they can cause [issues when interacting with external code](https://github.com/tc39/proposal-regexp-legacy-features/blob/master/subclass-restriction-motivation.md#legacy-static-properties-regexp1-etc)!
```
